### PR TITLE
feat(regl-renderer): refactored and corrected conversions of solids to entities, supporting HUGE solids

### PR DESCRIPTION
**Pull request by z3dev**
These changes enhance / correct the conversion of solids to visual entities, allowing HUGE solids to be rendered properly.

_NOTE: There's a simple example in the linked issue, but beware... it may crash the graphics card when executed in order versions._

There are additional changes to the demo applications in order to improve documentation.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

I threw everything that I have at these changes, including previous tests, my designs, etc.
Also tested with Firefox, Chrome, and Safari.

### CODE REVIEW

**Comment by z3dev** [comment]
@SimonClark please review.

there's some big refactoring of code, so it may be better just to read through the new code. all comments welcome.

FYI, there's still more to do to improve performance, etc.

**Review by z3dev — COMMENTED** [review]
Reviewed again. Good changes.
There’s still more to do though...
- improve direct conversions into buffers
- Implement DrawSegments and rework code

File: packages/utils/regl-renderer/demo.html [hunk 1]
```diff
@@ -4,19 +4,53 @@
...
+        height: 15cm;
         margin: 0;
         outline: 1px solid black;
       }
     </style>
 </head>
 <body>
-  <script language="javascript" src="./dist/jscad-regl-renderer.min.js"></script>
+  <script language="javascript" src="https://unpkg.com/@jscad/modeling" id="MODELING"></script>
+  <script language="javascript" src="./dist/jscad-regl-renderer.min.js" id="RENDERING"></script>
```
Comment (mister-jscad) [inline packages/utils/regl-renderer/demo.html:16]: Why not use the library from unpkg.com?

**Review by mister-jscad — APPROVED** [review]
Changes look fine.

Just one small question about demo.html

**Pull request merged**